WallPaperDev/WallPaperDev/Views/Controllers/DetailGoalViewController.swift

```diff
@@ -9,46 +9,108 @@
+    blueButton.addTarget(self, action: #selector(updateTapped), for: .touchUpInside)
+  }
+}
 
+// MARK: - UI setup function
+extension DetailGoalViewController {
+  
   private func configTextFields(goal: Goal) {
     guard let unwrappedName = goal.name,
       let unwrappedDescription = goal.summary else { return }
     createGoalView.goalNameTextField.text = unwrappedName
     createGoalView.goalDescriptionTextView.text = unwrappedDescription
     createGoalView.goalDescriptionTextView.textColor = ApplicationDependency.manager.currentTheme.colors.black
   }
-
-  override func setupNavBar() {
-    navigationItem.title = Localized.string("goal_details_title")
-    navigationController?.navigationBar.largeTitleTextAttributes = navigationController?.navigationBar.configLargeText(length: Localized.string("goal_details_title"))
-    navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .trash, target: self, action: #selector(deleteTapped))
+  
+  private func setupMilestoneLabel() {
+    mileStoneLabel.text = Localized.string("milestone_title")
+    mileStoneLabel.sizeToFit()
+    view.addSubview(mileStoneLabel)
+    mileStoneLabel.leadingAnchor.constraint(equalTo: createGoalView.goalNameLabel.leadingAnchor).isActive = true
+    mileStoneLabel.topAnchor.constraint(equalTo: createGoalView.goalDescriptionTextView.bottomAnchor, constant: 10).isActive = true
   }
-
-  private func configButton() {
-    createGoalView.createButton.setTitle(Localized.string("update_action"), for: .normal)
-    createGoalView.createButton.removeTarget(nil, action: nil, for: .allEvents)
-    createGoalView.createButton.addTarget(self, action: #selector(updateTapped), for: .touchUpInside)
+  
+  private func setupAddButton() {
+    addButton.translatesAutoresizingMaskIntoConstraints = false
+    addButton.tintColor = ApplicationDependency.manager.currentTheme.colors.navBarBlue
+    addButton.contentVerticalAlignment = .fill
+    addButton.contentHorizontalAlignment = .fill
+    addButton.setImage(UIImage(systemName: "plus.circle")?.withRenderingMode(.alwaysTemplate), for: .highlighted)
+    addButton.setImage(UIImage(systemName: "plus.circle.fill")?.withRenderingMode(.alwaysTemplate), for: .normal)
+    view.addSubview(addButton)
+    NSLayoutConstraint.activate([
+      addButton.heightAnchor.constraint(equalTo: mileStoneLabel.heightAnchor, multiplier: 1),
+      addButton.widthAnchor.constraint(equalTo: mileStoneLabel.heightAnchor, multiplier: 1),
+      addButton.topAnchor.constraint(equalTo: mileStoneLabel.topAnchor),
+      addButton.trailingAnchor.constraint(equalTo: createGoalView.goalDescriptionTextView.trailingAnchor)
+    ])
+    addButton.addTarget(self, action: #selector(showMilestoneprompt), for: .touchUpInside)
+  }
+  
+  private func setupTable() {
+    view.addSubview(milestonesTableView)
+    milestonesTableView.dataSource = self
+    milestonesTableView.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate([
```

Comment:
Have you ever used SnapKit? Personally I find it a bit less verbose than NSLayouts as it is a syntactical sugar tool :)

Reply:
You really like Snapkit huh ? 😉

I will add it to the backlog to do install it after the release since I don't know how long it will take to convert `NS:Layouts` to Snapkit.

Reply:
😆 haha did not even catch myself always mentioning it - for sure